lib/ed25519.js

```diff
@@ -36,6 +44,165 @@ ed25519.constants.SEED_BYTE_LENGTH = 32;
 ed25519.constants.SIGN_BYTE_LENGTH = 64;
 ed25519.constants.HASH_BYTE_LENGTH = 64;
 
+/**
+ * ASN1 public key validator for importing ed25519 keys in DER format.
+ * @type {Object}
+*/
+var ed25519PublicKeyValidator = forge.ed25519.publicKeyValidator = {
+  name: 'ed25519PublicKey',
+  tagClass: asn1.Class.UNIVERSAL,
+  type: asn1.Type.SEQUENCE,
+  constructed: true,
+  composed: true,
+  value: [{
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.SEQUENCE,
+    constructed: true,
+    composed: true,
+    value: [{
+      tagClass: asn1.Class.UNIVERSAL,
+      type: asn1.Type.OID,
+      constructed: false,
+      composed: false,
+      capture: 'ed25519Oid'
+    }]
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.BITSTRING,
+    constructed: false,
+    composed: true,
+    captureBitStringValue: 'ed25519PublicKey'
+  }]
+};
+
+/**
+ * ASN1 private key validator for importing ed25519 keys in DER format.
+ * @type {Object}
+*/
+var ed25519PrivateKeyValidator = forge.ed25519.privateKeyValidator = {
+  name: 'ed25519PrivateKey',
+  tagClass: asn1.Class.UNIVERSAL,
+  type: asn1.Type.SEQUENCE,
+  constructed: true,
+  composed: true,
+  value: [{
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.INTEGER,
+    constructed: false,
+    composed: false,
+    capture: 'privateKeyZero' // This value should always be 0.
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.SEQUENCE,
+    constructed: true,
+    composed: true,
+    value: [{
+      tagClass: asn1.Class.UNIVERSAL,
+      type: asn1.Type.OID,
+      constructed: false,
+      composed: false,
+      capture: 'ed25519Oid'
+    }]
+  }, {
+    tagClass: asn1.Class.UNIVERSAL,
+    type: asn1.Type.OCTETSTRING,
+    constructed: false,
+    composed: false,
+    capture: 'ed25519PrivateKey'
+  }]
+};
+
+/**
+ * Used to extract the asn1 from DER formatted key.
+ *
+ * @param {NativeBuffer} keyBuffer - A DER formatted key.
+ *
+ * @return {Object} An ASN1 object.
+ */
+function getKeyBytes(keyBuffer) {
+  var binary = forge.util.binary.raw.encode(keyBuffer);
+  return forge.asn1.fromDer(binary);
+}
+
+/**
+ * Given a DER formatted key this function will extract the raw key value.
+ *
+ * @param {Object} options - The options for this operation.
+ * @param {NativeBuffer} options.keyBuffer -
+ * The result of exporting the DER key.
+ * @param {ed25519PrivateKeyValidator|ed25519PublicKeyValidator}
+ * options.validator - A validator to check the exported key.
+ * @param {Object} [options.capture = {}] - A collection of name value pairs
+ * captured by the validator.
+ * @param {'ed25519PrivateKey'|'ed25519PublicKey'} options.keyName -
+ * The name of the key captured by the validator.
+ *
+ * @returns {string} - A binary string or the key value.
+ */
+ed25519.extractKey = function(options) {
```

Comment:
Let's not expose this method publicly yet, it will enable us to get this PR in sooner. There's a question about what the name and shape of this API should be or whether it should be public at all.